smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/AddApiKeySource.java

```diff
@@ -26,7 +27,7 @@
     private static final Logger LOGGER = Logger.getLogger(AddApiKeySource.class.getName());
 
     @Override
-    public OpenApi after(Context context, OpenApi openApi) {
+    public OpenApi after(Context<? extends Trait> context, OpenApi openApi) {
```

Comment:
Note: the `Context` that's passed around is now generic over the protocol trait being converted. For things where this doesn't matter, we use `<? extends Trait>`.

The other thing to keep in mind: the changes to the OpenAPI converters and API Gateway converters should result in almost identical OpenAPI schemas (the only thing that changes is the arbitrary names given to auth schemes)